### PR TITLE
working on making the makefile cross platform

### PR DESCRIPTION
**Pull request by whyrusleeping**
@jbenet this works on my osx and on my arch install.

### CODE REVIEW

File: udt4/src/Makefile [hunk 1]
```diff
@@ -18,15 +18,15 @@ ifeq ($(arch), POWERPC)
...
+ifeq ($(arch), sparc64)
    CCFLAGS += -DSPARC
 endif
 
-ifeq ($(arch), IA64)
+ifeq ($(arch), ia64)
    CCFLAGS += -DIA64
 endif
 
-ifeq ($(arch), AMD64)
```
Comment (jbenet) [inline udt4/src/Makefile]: is this correct across archs? is `amd64` not used in some?

maybe use something like in https://github.com/jbenet/go-env-sh/blob/master/go-platform

**Comment by jbenet** [comment]
i think we should change all their makefiles in one go, no? worried things will only sort of work?

File: udt4/src/Makefile (hunk 1 above)
Comment (whyrusleeping) [inline udt4/src/Makefile]: I used this (albeit incomplete) list here https://en.wikipedia.org/wiki/Uname

**Comment by whyrusleeping** [comment]
which other makefiles? the one in the top 'udt4' dir just runs things in the lower makefile.

File: udt4/src/Makefile (hunk 1 above)
Comment (jbenet) [inline udt4/src/Makefile]: FreeBSD uses `amd64`.

**Comment by jbenet** [comment]
the `app` ones? meh, ok

File: udt4/src/Makefile (hunk 1 above)
Comment (whyrusleeping) [inline udt4/src/Makefile]: alright, updated to catch that one.

Comment (jbenet) [inline udt4/src/Makefile]: ok sounds good. we can keep adding them here as needed i guess.

**Pull request merged**